protocol/bc/transaction.go

```diff
@@ -19,35 +18,43 @@ const CurrentTransactionVersion = 1
 // Tx holds a transaction along with its hash.
 type Tx struct {
 	TxData
-	TxHashes `json:"-"`
+	*TxEntries `json:"-"`
 }
 
 func (tx *Tx) UnmarshalText(p []byte) error {
 	if err := tx.TxData.UnmarshalText(p); err != nil {
 		return err
 	}
 
-	hashes, err := ComputeTxHashes(&tx.TxData)
+	txEntries, err := ComputeTxEntries(&tx.TxData)
 	if err != nil {
 		return errors.Wrap(err)
 	}
 
-	tx.TxHashes = *hashes
+	tx.TxEntries = txEntries
 	return nil
 }
 
+func (tx *Tx) IssuanceHash(n uint32) Hash {
```

Comment:
This parameter type should be int so we don't have to do conversions at the call sites. This was one of the NCC group recommendations from their first audit.